encrypted-media/scripts/playback-retrieve-persistent-license.js

```diff
@@ -0,0 +1,114 @@
+function runTest(config, testname) {
+
+    var testname = config.keysystem + ', retrieve persistent-license in new window, '
+                                    + /video\/([^;]*)/.exec( config.videoType )[ 1 ];
+
+    var configuration = {   initDataTypes: [ config.initDataType ],
+                            audioCapabilities: [ { contentType: config.audioType } ],
+                            videoCapabilities: [ { contentType: config.videoType } ],
+                            sessionTypes: [ 'persistent-license' ] };
+
+
+    async_test( function( test ) {
+        var _video = config.video,
+            _mediaKeys,
+            _mediaKeySession,
+            _mediaSource,
+            _sessionId;
+
+        function onFailure(error) {
+            forceTestFailureFromPromise(test, error);
+        }
+
+        function onEncrypted(event) {
+            assert_equals(event.target, _video);
+            assert_true(event instanceof window.MediaEncryptedEvent);
+            assert_equals(event.type, 'encrypted');
+
+            waitForEventAndRunStep('message', _mediaKeySession, onMessage, test);
+            _mediaKeySession.generateRequest(   config.initDataType || event.initDataType,
+                                                config.initData || event.initData ).then( function() {
+                _sessionId = _mediaKeySession.sessionId;
+            }).catch(onFailure);
+        }
+
+        function onMessage(event) {
+            assert_equals( event.target, _mediaKeySession );
+            assert_true( event instanceof window.MediaKeyMessageEvent );
+            assert_equals( event.type, 'message');
+
+            assert_in_array(  event.messageType, [ 'license-request', 'individualization-request' ] );
+
+            config.messagehandler( config.keysystem, event.messageType, event.message ).then( function( response ) {
+                _mediaKeySession.update( response )
+                .catch(onFailure);
+            });
+        }
+
+        function onPlaying(event) {
+            // Not using waitForEventAndRunStep() to avoid too many
+            // EVENT(onTimeUpdate) logs.
+            _video.addEventListener('timeupdate', onTimeupdate, true);
+        }
+
+        function onTimeupdate(event) {
+            if ( _video.currentTime > ( config.duration || 5 ) ) {
+
+                _video.removeEventListener('timeupdate', onTimeupdate );
+
+                _video.pause();
+
+                _mediaKeySession.closed.then( test.step_func( onClosed ) );
+
+                _mediaKeySession.close();
+            }
+        }
+
+        function onClosed() {
+            _video.src = "";
```

Comment:
What is the purpose of these two lines?

If the purpose here is to destroy the MediaKeys (CDM instance) before loading another one:
1. Probably add a comment as such.
2. Put everything in a .then(). This will at least wait until the media player is no longer using the CDM.
3. Clear _mediaKeys and _mediaKeySession.
Still, there is no guarantee that the underlying CDM [instance] or session has been destroyed due to garbage collection.